lib/Transforms/Scalar/ScalarReplAggregatesHLSL.cpp

```diff
@@ -1828,13 +1828,13 @@ bool SROAGlobalAndAllocas(HLModule &HLM, bool bHasDbgInfo) {
       bool SROAed = false;
       if (GlobalVariable *NewEltGV = dyn_cast_or_null<GlobalVariable>(
               TranslatePtrIfUsedByLoweredFn(GV, typeSys))) {
+        GVDbgOffset dbgOffset = GVDbgOffsetMap[GV];
+        GVDbgOffsetMap[NewEltGV] = dbgOffset;
```

Comment:
Good change, but maybe should remove GV from map in case GV is removed in the following code.